src/renderer/pages/Operations/common/utils.ts

```diff
@@ -218,3 +236,39 @@ export const getSignatoryName = (
 
   return toAddress(signatoryId, { chunk: 5, prefix: addressPrefix });
 };
+
+export const getSignatoryAccounts = (
+  accounts: Account[],
+  wallets: Wallet[],
+  events: MultisigEvent[],
+  signatories: Signatory[],
+  chainId: ChainId,
+) => {
+  const walletsDict = keyBy(wallets, 'id');
+
+  return signatories.reduce((accum: Account[], signatory) => {
+    const filteredAccounts = accounts.filter(
+      (a) => a.accountId === signatory.accountId && !events.some((e) => e.accountId === a.accountId),
+    );
+
+    const signatoryAccount = filteredAccounts.find((a) => {
+      const isCurrentChain = accountUtils.isChainIdMatch(a, chainId);
+      const walletType = walletsDict[a.walletId].type;
+      // TODO add wallet connect
+      const isCorrectWalletType = walletType === WalletType.SINGLE_PARITY_SIGNER;
```

Comment:
maybe add function here too like `isChainIdMatch`?

Reply:
added to wallet utils